### ISSUE TITLE
await channel.publish do not sent message with process.exit

### ISSUE THREAD

**Issue by ThomasDupont** [squaremo/amqp.node#581]
Hello

I past a lot of working hours on a channel publish issue.
I'm using amqplib 0.5.6 with node12 as a CLI (node sendMsg.js ...) in this case

My code look like:

```javascript
amqplib.connect('url', {
     verifyPeer: false,
}).then(async (conn) => {
    const channel = await conn.createChannel();
    const response = await channel.publish('exchange', 'routingKey', 'message', headers);

    console.log(response);
    process.exit(0);
});
```

The message is noted as sent but never received

The same code without the process.exit(0) works fine. There are, maybe, inside the lib a no 'awaiter' promise. I didn't investigate yet.

**Comment by cressie176** [comment]
Hi @ThomasDupont, `channel.publish` does not work in the way you expect.

> NOTE: Does not return a promise in the promises API; see [flow control](https://www.squaremobius.net/amqp.node/channel_api.html#flowcontrol)

TL;DR use confirm channels and pass a callback to `channel.publish`

**Comment by ThomasDupont** [comment]
Hello

Thanks, I will have a look and post the code I used to solve my issue

**Comment by USHIERU** [comment]
@ThomasDupont How did you solve it?

**Comment by ThomasDupont** [comment]
Yes, with channel.publish

```javascript
new Promise((res, rej) => this.channel.publish(
            exchange ? exchange : context.exchange,
            routingKey ? routingKey : context.routingKey,
            context.message,
            defaultHeaders,
            err => {
                if (err !== null) {
                    return rej(err);
                }
                return res(true);
            })
        );
```

Channel gave by
```javascript
amqplib.connect(
        `coURL`,
        {
            verifyPeer: false,
        },
    ).then(async (conn) => ({
            connection: conn,
            channel: await conn.createConfirmChannel(),
        })
    );
```
